aspnet/mvc/views/razor.rst

```diff
@@ -1,8 +1,740 @@
 
-.. include:: /../common/stub-notice.txt
+.. contents:: Sections:
+  :local:
+  :depth: 1
+  
+Rendering HTML
+--------------
 
-.. _issue: https://github.com/aspnet/Docs/issues/130
+The default Razor language is HTML. Rendering HTML from Razor is no different than in an HTML file. A Razor file with the following markup:
+
+.. code-block:: none
+
+  <p>Hello World</p> 
+
+Is rendered unchanged as ``<p>Hello World</p>`` by the server.
+
+Rendering server code
+----------------------
+
+Razor supports C# and uses the ``@`` symbol to transition from HTML to C#. 
+Razor can transition from HTML into C# or into Razor specific markup. When an ``@`` symbol is followed by a :ref:`Razor reserved keyword <Razor-reserved-keywords-label>` it transitions into Razor specific markup, otherwise it transitions into plain C# . 
+
+Razor expressions
+---------------------
+
+Implicit Razor expressions start with ``@`` followed by C# code. For example:
+
+.. code-block:: none
+
+  <p>@DateTime.Now</p>
+  <p>@DateTime.IsLeapYear(2016)</p>
+
+Implicit expressions generally cannot contain spaces. For example, in the code below, one week is not subtracted from the current time:
+
+.. literalinclude:: razor/sample/Views/Home/Contact.cshtml
+  :language: html
+  :start-after: @* End of greeting *@ 
+  :end-before: @*Add () to get correct time.*@
+
+Which renders the following HTML:
+
+.. code-block:: none
+ 
+  <p>
+    Last week: 7/7/2016 4:39:52 PM - TimeSpan.FromDays(7)
+  </p>
+
+Adding parenthesis fixes the problem:
+
+.. literalinclude:: razor/sample/Views/Home/Contact.cshtml
+  :language: html
+  :start-after: @*Add () to get correct time.*@
+  :end-before: @*End of correct time*@
+
+Which renders the following HTML:  
+
+.. code-block:: none
+
+  <p>
+    Last week: 6/30/2016 4:39:52 PM
+ </p>
+ 
+Explicit Razor code blocks start with ``@`` and are enclosed by ``()`` or ``{}``. For example:
+
+.. literalinclude:: razor/sample/Views/Home/Contact.cshtml
+  :language: html
+  :start-after: }
+  :end-before: @* End of greeting *@ 
+
+Renders this HTML markup:
+
+.. code-block:: none
+
+  <!-- Single statement blocks, explicit.  -->
+  
+  <!-- Multi-statement block, explicit.  -->
+    <p>The greeting is :<br /> Welcome! Today is: Monday -Leap year: True</p>
+    <p>The value of your account is: 7 </p>
+    <p>The value of myMessage is: Hello Hello World</p>
+
+With the exception of the C# ``await`` keyword implicit expressions must not contain spaces. For example, you can intermingle spaces as long as the C# statement has a clear ending:
+
+.. code-block:: none
+
+  <p>@await DoSomething("hello", "world")</p>
+
+HTML containing ``@`` symbols may need to be escaped with a second ``@`` symbol. For example:
+  
+.. code-block:: none
+
+ <p>@@Username</p> 
+ 
+would render the following HTML:
+
+.. code-block:: none
+ 
+ <p>@Username</p> 
+
+.. _razor-email-ref:
+
+``@`` in an email alias
```

Comment:
This still seems like it's own section. Move the contents of this, and the above `@@` escaping into an FYI like portion of the rendering server code section.